src/space.cpp

```diff
@@ -579,6 +579,13 @@ static argval_t do_space(chunk_t *first, chunk_t *second, int &min_sp, bool comp
          log_rule("FORCE");
          return(AV_FORCE);
       }
+      // U0_sp-before_ellipsis = AV_IGNORE returning AV_ADD which is causing issue 1291.
+      // To fix issue 1291
+      if (first->type == CT_WORD)
```

Comment:
@KalyanaChakravarthi I'd like you to test this bug with upstream because Guy says that it's fixed without your change. I'd prefer to take the upstream change (as we are already taking it) than adding a special handling here which looks a bit brittle.

Reply:
@mihaipopescu we tested the issue with upstream code which is working fine.
Removed my code changes in PR 57 as part of down merge.